chainer/functions/connection/convolution_nd.py

```diff
@@ -0,0 +1,354 @@
+import numpy
+import operator
+
+from functools import reduce
+from six import moves
+
+from chainer import cuda
+from chainer import function
+from chainer.functions.connection import convolution_2d
+from chainer.utils import conv
+from chainer.utils import conv_nd
+from chainer.utils import type_check
+
+
+if cuda.cudnn_enabled:
+    cudnn = cuda.cudnn
+    libcudnn = cuda.cudnn.cudnn
+    _cudnn_version = libcudnn.getVersion()
+    _fwd_pref = libcudnn.CUDNN_CONVOLUTION_FWD_SPECIFY_WORKSPACE_LIMIT
+    if _cudnn_version >= 4000:
+        _bwd_filter_pref = \
+            libcudnn.CUDNN_CONVOLUTION_BWD_FILTER_SPECIFY_WORKSPACE_LIMIT
+        _bwd_data_pref = \
+            libcudnn.CUDNN_CONVOLUTION_BWD_DATA_SPECIFY_WORKSPACE_LIMIT
+
+
+def _tuple(x, N):
```

Comment:
I think we should move to `utils` module as we define this function locally in another module.